### DISCUSSION TITLE
osu!lazer not starting on Arch Linux (X11, Nvidia GT 1030)

### DISCUSSION THREAD

**Discussion by ungeskriptet** [ppy/osu#21239] [Q&A]
### Type

Crash to desktop

### Bug description

After updating my system, osu!lazer no longer starts up and is stuck on a black screen. I tried rebooting and using older AppImage versions of osu!lazer but both options did not work. I use Arch Linux with GNOME (X11) on a Nvidia GT 1030.

### Screenshots or videos

![blackscreen :(](https://user-images.githubusercontent.com/40729975/201741110-3a9b63d6-1c1a-49ba-979f-9769d54d0eee.png)

### Version

2022.1101.0 and older

### Logs

[runtime.log](https://github.com/ppy/osu/files/10006157/runtime.log)
[performance.log](https://github.com/ppy/osu/files/10006158/performance.log)
[network.log](https://github.com/ppy/osu/files/10006159/network.log)
[input.log](https://github.com/ppy/osu/files/10006160/input.log)

**Comment by bdach** [comment]
Please post output of `dmesg`.

**Reply by ungeskriptet** [reply]
I'm not sure how my dmesg can help. This is more of an user space problem than kernel space problem 🤔

**Reply by bdach** [reply]
I'm not sure how you can assert that, but if you're not willing to provide extra information to attempt to debug this issue, then you're on your own.

**Reply by koraynilay** [reply]
Hi, I (probably) have the same issue, I'm also on Arch Linux (X11, i3-gaps) with kernel `6.0.8-arch1-1` and version `2022.1101.0` (using [`osu-lazer-bin`](https://aur.archlinux.org/packages/osu-lazer-bin)).

[database.log](https://github.com/ppy/osu/files/10024259/database.log)
[input.log](https://github.com/ppy/osu/files/10024260/input.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/10024261/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/10024264/network.log)
[performance.log](https://github.com/ppy/osu/files/10024265/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10024266/runtime.log)


And [my output of `sudo dmesg`](https://pastebin.com/ygpcryVg) (I removed all the `[UFW BLOCK]`, since there are a lot of them and when running `sudo dmesg --follow` with ufw disabled and then starting `osu-lazer` nothing pops up)

**Bug description:**
For me after starting `osu-lazer` the window opens, but sometimes it's just frozen right away, while other times the game loads but freezes after some seconds (the most I got was when I was able to enter solo mode and then it froze).
I don't know if it could be related, but I have around ~25GB of beatmaps (I don't know how to see how many I have).
I use `it_IT.UTF-8` as my locale.

**Hardware** (using neofetch):
**CPU:** Intel i5-6400 (4) @ 3.300GHz
**GPU:** NVIDIA GeForce GTX 1060 6GB
**GPU Driver:** NVIDIA 520.56.06
**Memory:** 15.57GiB
**Resolution:** 1920x1080 @ 144.00Hz
The game is installed on a SATA SSD

**Reply by koraynilay** [reply]
**Update:** I tried running it as `LC_ALL=C osu-lazer` and after only the first time (where it freezes right away and I have to kill it) it works.

These are the new logs (of the second launch, though it shouldn't matter):
[database.log](https://github.com/ppy/osu/files/10024640/database.log)
[input.log](https://github.com/ppy/osu/files/10024641/input.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/10024642/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/10024643/network.log)
[performance.log](https://github.com/ppy/osu/files/10024644/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10024645/runtime.log)

**Reply by GoldM-amberdrake** [reply]
Hello,

Can also confirm this on 2 machines with freshly updated Manjaro. It takes some time to launch the game and after it launches, only a black screen appears. The window freezes after couple more seconds.

Info about one of the machines (don't have access to the second one now, though):

OS: Manjaro Linux x86_64 
Kernel: 6.0.8-1-MANJARO 
Resolution: 1024x1280, 3840x2160, 1920x1080, 1920x1080 
DE: GNOME 43.1 
WM: Mutter
CPU: Intel i7-4790K (8) @ 4.600GHz 
GPU: NVIDIA GeForce GTX 1060 6GB 
GPU Driver: NVIDIA 520.56.06-13
Memory: 32042MiB

[runtime.log](https://github.com/ppy/osu/files/10025766/runtime.log)
[performance-update.log](https://github.com/ppy/osu/files/10025767/performance-update.log)
[performance-draw.log](https://github.com/ppy/osu/files/10025768/performance-draw.log)
[performance.log](https://github.com/ppy/osu/files/10025770/performance.log)
[network.log](https://github.com/ppy/osu/files/10025771/network.log)
[input.log](https://github.com/ppy/osu/files/10025772/input.log)
[database.log](https://github.com/ppy/osu/files/10025773/database.log)

Unfortunately doesn't work even when running with `LC_ALL=C osu-lazer`
It is the same with the AppImage and "osu-lazer-bin" AUR package.
Output from when launching Osu! from terminal:

```
[amberdrake@amb-manjaro-01 ~]$ LC_ALL=C osu-lazer
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
ioctl (GFEATURE): Connection timed out
/usr/bin/osu-lazer: line 2:  7160 Killed                  env OSU_EXTERNAL_UPDATE_PROVIDER=1 /opt/osu-lazer/osu.AppImage "$@"
```

`dmesg --follow` does not produce any additional output after Osu! is launched.

**Update:**
The issue persists even when running an older kernel - 5.19.17-2-MANJARO or a newer, experimental one - 6.1.0-1-MANJARO
**Update 2:**
Here's a list of packages that got updated. Might help. https://pastebin.com/hjGhNxrE

**Reply by GoldM-amberdrake** [reply]
It works on AMD hardware, though:

OS: Manjaro Linux x86_64 
Kernel: 6.0.8-1-MANJARO  
DE: GNOME 43.1 
WM: Mutter 
CPU: AMD Ryzen 7 4800H with Radeon Graphics (16) @ 2.900GHz 
GPU: AMD ATI Radeon RX 5500/5500M / Pro 5500M 
GPU: AMD ATI 07:00.0 Renoir 
Memory: 15388MiB 

I tried downgrading the NVIDIA driver at one of the affected machines, as it was one of the packages that got upgraded, but the issue still persists.

**Comment by GoldM-amberdrake** [comment]
As a temporary solution I rolled back all updates installed on that day and it works again.

```
$ grep -a upgraded /var/log/pacman.log| grep 2022-11-15 > /tmp/lastupdates.txt                                                              
$ awk '{print $4}' /tmp/lastupdates.txt > /tmp/lines1;awk '{print $5}' /tmp/lastupdates.txt | sed 's/(/-/g' > /tmp/lines2
$ paste /tmp/lines1 /tmp/lines2 > /tmp/lines
$ tr -d "[:blank:]" < /tmp/lines > /tmp/packages
$ cd /var/cache/pacman/pkg/
$ for i in $(cat /tmp/packages); do sudo pacman --noconfirm -U "$i"*; done
```

**Reply by GoldM-amberdrake** [reply]
**Update:**
I think I've got it. Narrowed it down to libx11. Confirmed working with "libx11-1.8.1-1". If you don't have it in you Pacman cache, you can download it here: https://archive.archlinux.org/packages/l/libx11/

**Reply by Urantij** [reply]
libx11-1.8.1-3 is the last working version. The next one is libx11-1.8.2-1, and it doesnt work.

**Comment by C9Glax** [comment]
Try [disabling wacom driver](https://github.com/ppy/osu/discussions/21526) (since that was my case).
